mne/viz/_mpl_figure.py

```diff
@@ -1567,6 +1567,12 @@ def _show_scalebars(self):
                 denom = 4 if self.mne.butterfly else 2
                 y = tuple(np.array([-1, 1]) / denom + offset)
                 self._draw_one_scalebar(x, y, this_type)
+                if self.mne.is_epochs:
+                    x = (self.mne.times[0], self.mne.times[0] +
+                         self.mne.boundary_times[1] / 2)
```

Comment:
if I'm reading this correctly, it looks like your scalebar is always 1/2 the duration of the epoch. Was that a considered choice, or a convenient one?  I would have expected the scalebar to always be a nice round number... something like "the largest duration `2**n * 5**m` that is smaller than the epoch duration".  I.e., 2, 5, 10, 20, 50, 100, 200, 500, 1000, etc (in milliseconds). I think this gets you close:

```py
_dur = np.diff(fig.mne.boundary_times[:2])[0]
bar_sizes = np.array([1, 2, 5]) * 10 ** np.floor(np.log10(_dur))
# followed by an np.searchsorted() to get the index of 
# the largest bar that is smaller than _dur
```

but I'm curious to hear if others have a different preference here.

Reply:
I chose based on the aesthetic of how the scale bar fit into the space. You think it should be smaller? Or you prefer it to be in ms?

Reply:
There are two separable issues here:

1. what duration should the bar represent?
2. should the number be displayed in seconds or milliseconds?

For point 1, I don't necessarily think it should be shorter (or longer). I think it should be a nice round number. Think about scale bars on maps, etc. They tell you how long 1 km is in the map scale; they don't tell you "half the map width corresponds to 252.47 km."

For point 2, I think given the time scales that most people deal with when epoching, displaying in ms makes sense; e.g., `50 ms` seems nicer than `0.05 s`. We can also make it responsive, so that if someone has 8-second-long epochs, the label can say e.g., `5 s`, but if the epoch is shorter than 1s then it displays the length in ms.

As far as aesthetics goes, you're probably right that it will look best if the scalebar is somewhere between, I dunno, 1/4 to 1/2 the width of the epoch on the plot (I just ballparked that without actually playing around with it, use whatever limits are actually aesthetically nice). Once you've decided what those limits are, you can tweak the `bar_sizes` candidates in my example snippet to make sure we end up with one in that range.

If you're sick of tweaking this, then what you have so far looks pretty nice and is not broken in any way (at least not that I've detected) so I'm OK to merge this and treat points 1 and 2 above as refinements to be done in future PRs (by you or someone else). Of course if I've convinced you, you're free to do them in this PR too. WDYT?

Reply:
I'd prefer to use it as it is for now and revisit in future PRs. Thanks.